### PR TITLE
[MLU] Fix Llama attrntion_mask in npu and mlu

### PR DESCRIPTION
**Pull request by DrownFish19**
### PR types

Bug fixes
### PR changes

Others
### Description

Fix Llama attrntion_mask in npu and mlu.

### CODE REVIEW

File: paddlenlp/transformers/llama/modeling.py [hunk 1]
```diff
@@ -1653,7 +1653,7 @@ def forward(
                 is_casual = True
             else:
                 is_casual = is_casual_mask(attention_mask)
-            if get_env_device() != "npu" or get_env_device() != "mlu":
+            if get_env_device() != "npu" and get_env_device() != "mlu":
```
Comment (ZHUI) [inline paddlenlp/transformers/llama/modeling.py]:
```suggestion
            if get_env_device() not in ["npu", "mlu"]:
```

Comment (ZHUI) [reply paddlenlp/transformers/llama/modeling.py]: 这样会不会好一点

Comment (DrownFish19) [reply paddlenlp/transformers/llama/modeling.py]: 嗯嗯，明确一些

**Review by ZHUI — APPROVED** [review]
LGTM

**Pull request merged**